### ISSUE TITLE
"GridLayoutManager.multipleSpanGrid" It takes too much time with a large data set

### ISSUE THREAD

**Issue by kjsolo** [fondesa/recycler-view-divider#130]
```kotlin
val dataList = MutableList<Any>(1000) { index ->
    A()
}

dataList.add(0, B())

val layoutManager = GridLayoutManager(requireActivity(), 3)

layoutManager.spanSizeLookup = object : GridLayoutManager.SpanSizeLookup() {
    override fun getSpanSize(position: Int): Int {
        return when (dataList.getItem(position)) {
            B -> 3
            else -> 1
        }
    }
}

requireActivity().dividerBuilder()
    .showFirstDivider()
    .showSideDividers()
    .size(8.dp)
    .asSpace()
    .build()
    .addTo(recyclerView)
```

**Comment by kjsolo** [comment]
I found ways to improve, but there's still a little bit of stuck.

Cache the span size:

```kotlin
class SpanSizeUtils {

    private val spanTypeToSizeList = SparseIntArray()
    var typeToSizeRule: ((index: Int, item: Any) -> Int?)? = null

    fun update(dataList: List<Any>) {
        spanTypeToSizeList.clear()
        dataList.forEachIndexed { index, item ->
            val size = typeToSizeRule?.invoke(index, item)
            if (size != null) {
                spanTypeToSizeList.put(index, size)
            }
        }
    }

    fun getSpanSize(position: Int): Int {
        return spanTypeToSizeList.get(position, 1)
    }

}
```

**Comment by fondesa** [comment]
This will be included in the next release, now, the entire `Grid` is cached. My benchmarks showed a big performance improvement. The cache is invalidated automatically so from the next release, you will (hopefully) notice the performance improvements without changing anything. Thanks for the idea!

**Comment by fondesa** [comment]
Released in the version `3.4.0`. Thanks again for the suggestion!
